extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -43,19 +43,29 @@ const RESET_SCROLL_REFERENCE_POINT_WAIT_MS = 200;
  * @template T
  */
 function ScrollerWithRef(
-  {children, loop, restingIndex, setRestingIndex, snap},
+  {
+    advanceCount,
+    children,
+    loop,
+    restingIndex,
+    setRestingIndex,
+    snap,
+    visibleCount,
+  },
   ref
 ) {
   // We still need our own ref that we can always rely on to be there.
   const containerRef = useRef(null);
+  const advance = (by) => {
+    const container = containerRef.current;
+    const slideWidth = container./* OK */ offsetWidth / visibleCount;
+    // Modify scrollLeft is preferred to `setRestingIndex` to enable smooth scroll.
+    // Note: `setRestingIndex` will still be called on debounce by scroll handler.
+    container./* OK */ scrollLeft += slideWidth * by;
+  };
   useImperativeHandle(ref, () => ({
-    // Expose "advance" action for navigating between slides by the given quantity of slides.
-    advance: (by) => {
-      const container = containerRef.current;
-      // Modify scrollLeft is preferred to `setRestingIndex` to enable smooth scroll.
-      // Note: `setRestingIndex` will still be called on debounce by scroll handler.
-      container./* OK */ scrollLeft += container./* OK */ offsetWidth * by;
-    },
+    next: () => advance(advanceCount),
```

Comment:
I think this imperative handle needs to have deps passed. I checked and it's indeed recreated on each render unnecessary.

Reply:
Done, PTAL